### DISCUSSION TITLE
drawing large molecule in Ketcher

### DISCUSSION THREAD

**Discussion by pa-nana** [epam/ketcher#1656] [Q&A]
Dear Ketcher team,

We are using Ketcher (standalone, 2.4) for drawing the structure and getting the mole file (in V3000 format). Recently some users start to draw a quite large molecule, and we face the issue while saving as a mol file.

Our user was using another drawing tool and then he wanted to open the mol file in Ketcher 2. I tried this on the Ketcher 2 demo site using the "open from file" function. mol files can be opened. However, when I pressed "Add to canvas", the system showed the processing status and did not stop ..... . note: I waited about 10 minutes.

As the mol file is quite large (around 4.5MB), I've attached the structure diagram below

![large molecule](https://user-images.githubusercontent.com/87416818/172405947-e6832077-989e-4736-bb52-42eb1ef00d99.png)

I tried to draw the same structure in Ketcher 2 but could not finish it because there is a limit to the number of S-groups (between 1 and 1000). The structure is as follows

![s_groups 1000](https://user-images.githubusercontent.com/87416818/172410286-62904823-2481-482a-8640-2d37ae610016.png)

Therefore, I would like to know:

1. Will it help if I use the remote version instead of the stand-alone version?
2. Why do we have the limit on the number of S-groups (between 1 and 1000)?

Thank you.

**Comment by AndreiMazol** [comment]
Dear @CLLPCH ,

Thank you for your interest to Ketcher.

Please find answers below:

1. Unfortunately I guess that switching on remote version  won't fix your issue. If it is possible please share your large file to test it locally. we will try to find a solution and deliver it as a part of future releases.
2. This limitation is deprecated and will be deleted in scope of https://github.com/epam/ketcher/issues/1657 and will be delivered as a part of release 2.5.1 (in 2 weeks)

Best regards,
Ketcher team

**Reply by pa-nana** [reply]
Hi @AndreiMazol ,

Thanks for the fast response.

Yes, you're right about the using remote or stand-alone version. I've tried the remote version (demo site) and got the same result as stand-alone version.

The mol file is attached [large 10000.zip](https://github.com/epam/ketcher/files/8860169/large.10000.zip)

Thank you.
